### PR TITLE
Add lambda + api gateway error status codes article

### PR DESCRIPTION
**Pull request by kizza**

### CODE REVIEW

File: technologies/lambda/api-gateway-errors.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+}
+```
+...will result with the following response _(including the http status for clarity)_
+```js
+Status: 200
+{
+  "errorMessage": "Invalid data",
+  "errorType": "Error",
+  "stackTrace": [
+    "exports.handler (/var/task/index.js:11:17)"
```
Comment (bradparker) [inline technologies/lambda/api-gateway-errors.md:35]: should this be `exports.myHandler` ?

Comment (bradparker) [inline technologies/lambda/api-gateway-errors.md:35]: Also for the one export you could choose to use `module.exports`... if you like.

Comment (kizza) [inline technologies/lambda/api-gateway-errors.md:35]: I like the way you think @bradparker - yes I will go with module.exports

File: technologies/lambda/api-gateway-errors.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
+# Custom error codes with Lambda and API Gateway
+
+Combining Amazon's Lambda and API Gateway is an effective means of creating a robust and cost effective api.
+
+Lambda functions contains two primary methods called for success or failure but one quickly discovers  that ```context.fail()``` still returns a successful 200 response.  Ultimately both code paths below return 200.
```
Comment (bradparker) [inline technologies/lambda/api-gateway-errors.md]: "Lambda functions contain" ?

"Lambda functions are passed a `context` object containing both a `succeed` and a `fail` method. Both methods are for returning a value from your lambda process, on success or failure. You might expect `failure` to send a response with a 4XX or 5XX status code, unfortunately you'd be wrong." ... ?

File: technologies/lambda/api-gateway-errors.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+  "errorMessage": "A different error",
+  "errorType": "Error",
+  "stackTrace": [
+    "exports.handler (/var/task/index.js:11:17)"
+  ]
+}
+```
+
+### Response mapping templates
+As a final touch let's remove the ```stackTrack``` property from our error response (as for api purposes it's more information than we may wish to provide).  One-by-one expand each regex to reveal **Mapping templates** and enter ```application/json``` (don't forget the little 'tick' to save). Clicking your newly created mapping template will show **Output passthrough** towards the right (which currently allows the full error response through).  Click the edit button next to it and select **Mapping template** from the dropdown. The below json will specify your output pulling in the original data from the response but omitting the stackTrace.
```
Comment (bradparker) [inline technologies/lambda/api-gateway-errors.md]: `stackTrace` ?

**Comment by bradparker** [comment]
This is gonna be sooo helpful for future lambda-ers. I can only imagine the confusion you felt when seeing a 200 status on something passed to `context.fail`.

**Comment by kizza** [comment]
lol. Cheers - it feels like a dry topic to me - but yes, dave and I hit our heads against the desk for quite a while before the solution came through (and all the other tutorial go down rabbit trails)

File: technologies/lambda/api-gateway-errors.md (hunk 2 above)
Comment (kizza) [inline technologies/lambda/api-gateway-errors.md]: Super nice :)  :+1:

File: technologies/lambda/api-gateway-errors.md (hunk 3 above)
Comment (kizza) [inline technologies/lambda/api-gateway-errors.md]: No I definitely meant stackTrack. :) Nice catch!

File: technologies/lambda/api-gateway-errors.md (hunk 1 above)
Comment (kizza) [inline technologies/lambda/api-gateway-errors.md:35]: ... so I looked at the standard node setup boilerplate projects in aws, they all have the pattern of exporting a method called "handler" via exports.handler.  So thinking of using that as the default? (but still removing all the "myHandler" with "handler".  Thoughts?

Comment (bradparker) [inline technologies/lambda/api-gateway-errors.md:35]: Great, if there's something that people reading the docs / reading your article see consistently that's less to think about.

**Pull request merged**